### ISSUE TITLE
ERROR: Unable to download webpage: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED]

### ISSUE THREAD

**Issue by l0rddarkf0rce** [ytdl-org/youtube-dl#27793]
## Checklist

- [ ] I'm reporting a broken site support issue
- [x] I've verified that I'm running youtube-dl version **2021.01.08**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar bug reports including closed ones
- [x] I've read bugs section in FAQ

## Verbose log

```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '-f', '140', '-x', '--audio-format', 'mp3', 'https://youtu.be/Lnezh5wQQQ4']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.01.08
[debug] Python version 3.9.0 (CPython) - macOS-10.15.7-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.3.1-tessus
[debug] Proxy map: {}
[youtube] Lnezh5wQQQ4: Downloading webpage
ERROR: Unable to download webpage: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1122)> (caused by URLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1122)')))
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/youtube_dl/extractor/common.py", line 632, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/youtube_dl/YoutubeDL.py", line 2248, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 517, in open
    response = self._open(req, data)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 534, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/youtube_dl/utils.py", line 2734, in https_open
    return self.do_open(functools.partial(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 1345, in do_open
    raise URLError(err)
```

## Description

Not sure when, but between 20200101 and 20200113 the SSL failed certificate error creeped in again. adding '--no-check-certificate' seems to work, but this a temporary workaround rather than a fix. the command used is

`youtube-dl -f 140 -x --audio-format mp3 https://youtu.be/Lnezh5wQQQ4`

**Comment by Isaac-the-Man** [comment]
I am unable to reproduce this issue. Are you sure the url is playable in your browser?

**Comment by pixelomer** [comment]
I am also able to reproduce this on macOS Big Sur 11.1 (20C69). Python 3.9.1 was installed using MacPorts. I am able to play the same video in my browser with no issues.
```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://youtube.com/watch?v=<redacted>']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.01.08
[debug] Python version 3.9.1 (CPython) - macOS-11.1-x86_64-i386-64bit
[debug] exe versions: none
[debug] Proxy map: {}
[youtube] <redacted>: Downloading webpage
ERROR: Unable to download webpage: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1123)> (caused by URLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1123)')))
  File "~/Library/Python/3.9/lib/python/site-packages/youtube_dl/extractor/common.py", line 632, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "~/Library/Python/3.9/lib/python/site-packages/youtube_dl/YoutubeDL.py", line 2248, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 517, in open
    response = self._open(req, data)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 534, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "~/Library/Python/3.9/lib/python/site-packages/youtube_dl/utils.py", line 2734, in https_open
    return self.do_open(functools.partial(
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 1345, in do_open
    raise URLError(err)
```

**Comment by l0rddarkf0rce** [comment]
> I am unable to reproduce this issue. Are you sure the url is playable in your browser?

Yes... I just tried it and it plays just fine on a browser. The URL is not really relevant, it happens with every single URL that I try.

**Comment by Isaac-the-Man** [comment]
Hmm, that is strange. I just checked again and mine worked just fine. I'm on Win10 running python 3.4.4. Here's my log for reference.
```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-f', '140', '-x', '--audio-format', 'mp3', 'https://youtu.be/Lnezh5wQQQ4', '--verbose']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2021.01.08
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.19041
[debug] exe versions: ffmpeg 4.3.1-2021-01-01-full_build-www.gyan.dev, ffprobe 4.3.1-2021-01-01-full_build-www.gyan.dev
[debug] Proxy map: {}
[youtube] Lnezh5wQQQ4: Downloading webpage
[youtube] {18} signature length 108, html5 player 9f996d3e
[youtube] {136} signature length 108, html5 player 9f996d3e
[youtube] {247} signature length 108, html5 player 9f996d3e
[youtube] {398} signature length 108, html5 player 9f996d3e
[youtube] {135} signature length 104, html5 player 9f996d3e
[youtube] {244} signature length 108, html5 player 9f996d3e
[youtube] {397} signature length 104, html5 player 9f996d3e
[youtube] {134} signature length 108, html5 player 9f996d3e
[youtube] {243} signature length 108, html5 player 9f996d3e
[youtube] {396} signature length 108, html5 player 9f996d3e
[youtube] {133} signature length 104, html5 player 9f996d3e
[youtube] {242} signature length 108, html5 player 9f996d3e
[youtube] {395} signature length 108, html5 player 9f996d3e
[youtube] {160} signature length 108, html5 player 9f996d3e
[youtube] {278} signature length 108, html5 player 9f996d3e
[youtube] {394} signature length 108, html5 player 9f996d3e
[youtube] {140} signature length 108, html5 player 9f996d3e
[youtube] {249} signature length 108, html5 player 9f996d3e
[youtube] {250} signature length 108, html5 player 9f996d3e
[youtube] {251} signature length 108, html5 player 9f996d3e
[debug] Invoking downloader on 'https://r4---sn-ipoxu-u2xd.googlevideo.com/videoplayback?expire=1610657716&ei=VFsAYNbYLsTOqQH9hrPoDQ&ip=114.37.181.149&id=o-AP33dmb7Q9Pll4Rk1ShHnVb1sn0Wj8UzTTLvaTz709-e&itag=140&source=youtube&requiressl=yes&mh=tg&mm=31%2C29&mn=sn-ipoxu-u2xd%2Csn-un57en7l&ms=au%2Crdu&mv=m&mvi=4&pcm2cms=yes&pl=21&gcr=tw&initcwndbps=1108750&vprv=1&mime=audio%2Fmp4&ns=080Xx8Lyk7_8iNaUMfb3utYF&gir=yes&clen=3513581&dur=217.060&lmt=1578240876526034&mt=1610635742&fvip=4&keepalive=yes&c=WEB&txp=5531432&n=fZZ91gbdKDFnXg&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cgcr%2Cvprv%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpcm2cms%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRAIgA5hsJjEEaQwHzEHYXpJa0IiGCnmmleqR_DHLUP-uIkcCIBd7SWLWq0Y4rPBzcSimcSqBexVZ8pk5Bu6qNFn-z1Uf&sig=AOq0QJ8wRQIhAOJXSP_70sGFkadxWlxLR7yAYeP2qik7GcxHavv8A1e2AiBkkUitqNfphWi08HH4gKfnLdbgfU8JN-9Ri0tpHwO4ZQ==&ratebypass=yes'
[download] Destination: Enya - Only Time (8D AUDIO)-Lnezh5wQQQ4.m4a
[download] 100% of 3.35MiB in 00:00
[ffmpeg] Correcting container in "Enya - Only Time (8D AUDIO)-Lnezh5wQQQ4.m4a"
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:Enya - Only Time (8D AUDIO)-Lnezh5wQQQ4.m4a" -c copy -f mp4 "file:Enya - Only Time (8D AUDIO)-Lnezh5wQQQ4.temp.m4a"
[debug] ffmpeg command line: ffprobe -show_streams "file:Enya - Only Time (8D AUDIO)-Lnezh5wQQQ4.m4a"
[ffmpeg] Destination: Enya - Only Time (8D AUDIO)-Lnezh5wQQQ4.mp3
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:Enya - Only Time (8D AUDIO)-Lnezh5wQQQ4.m4a" -vn -acodec libmp3lame "-q:a" 5 "file:Enya - Only Time (8D AUDIO)-Lnezh5wQQQ4.mp3"
Deleting original file Enya - Only Time (8D AUDIO)-Lnezh5wQQQ4.m4a (pass -k to keep)
```

**Comment by SiegeSailor** [comment]
I am facing the same issue.

```Base
youtube-dl https://www.youtube.com/watch\?v\=o_TYVGp9n0s                                                                                                         
[youtube] o_TYVGp9n0s: Downloading webpage
ERROR: Unable to download webpage: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1124)> (caused by URLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1124)')))
```

I am able to play the video with my Chrome. However, the URL doesn't really matter, it happens to any URL.

#### Other Info

- macOS Catalina 10.15.7
- Python 3.8.6

**Comment by mungre** [comment]
I just had this same issue on a recent, up-to-date, Windows 10 install with a brand new install of Python 3.9.2.  I found a fix that doesn't seem to be documented yet.

youtube's root CA is "GlobalSign Root CA - R2".  The Windows certificate store has "GlobalSign Root CA - R1" but not R2.  Hence the error.

youtube and google must work in Microsoft Edge, so I guessed that Edge must be causing the certificate store to be updated somehow.  So I loaded Edge and entered google.com in the address bar.  The "GlobalSign Root CA - R2" certificate immediately appeared in the Windows certificate store.  And youtube-dl started working.  Hope this helps.  Thanks for youtube-dl!
